src/core/scalebar/qgsscalebarrenderer.cpp

```diff
@@ -61,20 +63,39 @@ void QgsScaleBarRenderer::drawDefaultLabels( QgsRenderContext &context, const Qg
     }
     else if ( segmentCounter != 0 && segmentCounter == nSegmentsLeft ) //reset label number to 0 if there are left segments
     {
-      currentLabelNumber = 0;
+      currentLabelNumber = 0.0;
     }
 
     if ( segmentCounter >= nSegmentsLeft )
     {
       currentNumericLabel = QString::number( currentLabelNumber / settings.mapUnitsPerScaleBarUnit() );
     }
 
-    if ( segmentCounter == 0 || segmentCounter >= nSegmentsLeft ) //don't draw label for intermediate left segments
+    bool drawZero = settings.labelHorizontalPlacement() != QgsScaleBarSettings::LabelCenteredSegment;
```

Comment:
Can you use switches wherever the placement options are handled? This will make it much easier to extend in future and see exactly where we need to consider new methods

Reply:
@nyalldawson , actually, this specific bool is in preparation to a [x] do not draw the zero label options, which would add another condition to that line.

Not sure where you'd want me to do a switch in here.

Reply:
Like this:


    bool drawZero = true;
    switch ( settings.labelHorizontalPlacement() )
    { 
        case QgsScaleBarSettings::LabelCenteredSegment:
             drawZero = false;
             break;

       //other cases
               break;
}